checks/raw/pinned_dependencies.go

```diff
@@ -283,27 +283,26 @@ var validateDockerfilesPinning fileparser.DoWhileTrueOnFileContent = func(
 		case len(valueList) == 1:
 			name := valueList[0]
 			pinned := pinnedAsNames[name]
-			if !pinned && !regex.MatchString(name) {
-				dep := checker.Dependency{
-					Location: &checker.File{
-						Path:      pathfn,
-						Type:      finding.FileTypeSource,
-						Offset:    uint(child.StartLine),
-						EndOffset: uint(child.EndLine),
-						Snippet:   child.Original,
-					},
-					Type: checker.DependencyUseTypeDockerfileContainerImage,
-				}
-				parts := strings.SplitN(name, ":", 2)
-				if len(parts) > 0 {
-					dep.Name = asPointer(parts[0])
-					if len(parts) > 1 {
-						dep.PinnedAt = asPointer(parts[1])
-					}
+
+			dep := checker.Dependency{
+				Location: &checker.File{
+					Path:      pathfn,
+					Type:      finding.FileTypeSource,
+					Offset:    uint(child.StartLine),
+					EndOffset: uint(child.EndLine),
+					Snippet:   child.Original,
+				},
+				Pinned: asBoolPointer(pinned || regex.MatchString(name)),
```

Comment:
I believe no, since in line 287 we create `dep` variable and in line 300 we modify `dep` to fill `PinnedAt`. I understand we do not need to modify this behavior. Does that make sense?